File: Python/BOJ/18258.py
import sys
from collections import deque

queue = deque()
n = int(sys.stdin.readline())

for _ in range(n):
    cmd = sys.stdin.readline().rstrip()
    if cmd[:4] == "push":
        queue.append(cmd.split()[1])
    elif cmd == "front":
        if queue:
            print(queue[0])
        else:
            print(-1)
    elif cmd == "back":
        if queue:
            print(queue[-1])
        else:
            print(-1)
    elif cmd == "pop":
        if queue:
            print(queue.popleft())
        else:
            print(-1)
    elif cmd == "size":
        print(len(queue))
    elif cmd == "empty":
        if queue:
            print(0)
        else:
            print(1)

File: C++/BOJ/18258.cpp
#include<iostream>
#include<deque>

using namespace std;

int n;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    cin>>n;
    deque<int> dq;
    for (int i = 0; i < n; i++)
    {
        string s;
        cin>>s;
        if (s == "push")
        {
            int num;
            cin>>num;
            dq.push_back(num);
        }
        else if (s == "pop")
        {
            if (dq.empty())
                cout<<"-1"<<"\n";
            else
            {
                cout<<dq.front()<<"\n";
                dq.pop_front();
            }
        }
        else if (s == "size")
            cout<<dq.size()<<"\n";
        else if (s == "empty")
            cout<<dq.empty()<<"\n";
        else if (s == "front")
        {
            if (dq.empty())
                cout<<"-1"<<"\n";
            else
                cout<<dq.front()<<"\n";
        }
        else if (s == "back")
        {
            if (dq.empty())
                cout<<"-1"<<"\n";
            else
                cout<<dq.back()<<"\n";
        }
    }
}